build/lib/compilation.ts

```diff
@@ -88,6 +88,8 @@ export function compileTask(src: string, out: string, build: boolean): () => Nod
 			generator.execute();
 		}
 
+		generateGitHubAuthConfig();
```

Comment:
@joaomoreno Would love if you could take a quick look at this - is this a reasonable spot in the build to put this?

For context, the github auth extension needs to have a client id and secret to perform OAuth. Currently, when registering an OAuth app on github, you can only specify one callback uri, so I'll have to have a registration for each scheme/VSO url. I plan to put these in keyvault and set them as environment variables when the build runs. I need to generate something the extension can read at runtime, which is what this does

Reply:
Won't these _secrets_ simply leak as soon as we release a version of Code containing this `config.json` file? If that's the case, why not simply inline them in a `config.json` file directly in our sources?

Reply:
Yes, they would then be available in the shipped product. This just adds a layer of obfuscation/makes it slightly harder to find them.

The security of OAuth comes from registering a specific callback uri, so even if someone had the other information, they wouldn't be able to receive the callback

Reply:
I really don't think this is a good idea. If these values will simply be public, then this is just making engineering more complicated for little gain. I would just inline them. You can even do this nicely by placing the `config.json` file in the distro.

Reply:
We talked about this some more in the Redmond standup yesterday - our feeling is that it's still better to avoid checking these values in. Putting things in keyvault is the approved way of storing info like this, I'm worried that as GitHub scans for creds, it would revoke these if it found them in source code.

Since we have a solution that is more secure than just checking them in, I think it's better to use it than to move to something less secure. I know these are being shipped anyway, and to actually use them requires something else to be majorly compromised, but I think this isn't too much overhead